src/lib.rs

```diff
@@ -3942,12 +3942,14 @@ impl Build {
         }
     }
     fn is_wasi_target(target: &str) -> bool {
-        const TARGETS: [&'static str; 5] = [
+        const TARGETS: [&'static str; 7] = [
             "wasm32-wasi",
             "wasm32-wasip1",
             "wasm32-wasip1-threads",
             "wasm32-wasip2",
             "wasm32-wasi-threads",
+            "wasm32-unknown-wasi",
+            "wasm32-unknown-unknown",
```

Comment:
With the additional of wasm32-unknown-unknown, I feel like I should rename this to `is_wasm_target` instead, but then this list here also doesn't cover all wasm targets.